backend/agentd/session.go

```diff
@@ -427,7 +430,28 @@ func (s *Session) stop() {
 		}
 	}()
 
+	// Send a close message to ensure the agent closes its connection if the
+	// connection is not already closed
+	if !s.conn.Closed() {
+		if err := s.conn.SendCloseMessage(); err != nil {
+			logger.WithError(err).Warning("unexpected error while sending a close message to the agent")
+		}
+	}
```

Comment:
This is a race condition, since the conn could close between checking it and the call to SendCloseMessage. Perhaps we should interrogate the error more carefully, or avoid logging it?

Reply:
Yeah, you're right, I don't think we really care about the error here